### ISSUE TITLE
Default destructor missing when using define_aggregate

### ISSUE THREAD

**Issue by ilmmatias** [bloomberg/clang-p2996#115]
According to [[meta.reflection.define.aggregate]](https://isocpp.org/files/papers/P2996R8.html#meta.reflection.define.aggregate-reflection-class-definition-generation), a default empty destructor should be provided by define_aggregate on unions containing alternatives with a non-trivial destructor, but that currently doesn't happen.
The same should also happen for alternatives with a non-trivial constructor (a default empty constructor should be given).

Without this, [3.9 A Simple Variant Type](https://isocpp.org/files/papers/P2996R8.html#a-simple-variant-type) doesn't compile when std::string is one of the pack parameters.
See: https://godbolt.org/z/3c1dqe6Ko

**Comment by katzdm** [comment]
You're right! Thanks for calling this out; I'd forgotten about this behavior.

**Comment by BaLiKfromUA** [comment]
Just for visibility, I am looking on this issue

**Comment by katzdm** [comment]
This is no longer part of P2996; it got split off into [P3074](https://wg21.link/p3074r7) which was separately accepted into C++26 during Wrocław. We removed it from P2996R10. If it gets implemented in upstream clang, we'll pull it into this fork during a merge.
